app/scripts/controllers/transactions/tx-state-manager.test.js

```diff
@@ -1074,6 +1075,136 @@ describe('TransactionStateManager', function () {
     });
   });
 
+  describe('#generateTxMeta', function () {
+    it('generates a txMeta object when supplied no parameters', function () {
+      // There are currently not safety checks for missing 'opts' but we should
+      // at least enforce txParams. This is done in the transaction controller
+      // before *calling* this method, but we should perhaps ensure that
+      // txParams is provided and validated in this method.
+      // TODO: this test should fail.
```

Comment:
Is this comment relevant?  Should we be making it fail now?

Reply:
Yes, this comment was meant to be leaving a TODO for myself as I explore optimizations / improvements to the tx-state-manager/transaction controller improvements.

It wasn't meant to be made to fail in this PR.